File: VB/MvcDashboardDataSources/Configuration/XpoDataSourceConfigurator.vb
Imports DevExpress.DashboardCommon
Imports DevExpress.DashboardWeb

Namespace MvcDashboardDataSources.Configuration
	Public Class XpoDataSourceConfigurator
		Public Shared Sub ConfigureDataSource(ByVal storage As DataSourceInMemoryStorage)
			' Registers an XPO data source.
			Dim xpoDataSource As New DashboardXpoDataSource("XPO Data Source")
			xpoDataSource.ConnectionStringName = "NWindConnectionStringSQLite"
			xpoDataSource.SetEntityType(GetType(Category))
			storage.RegisterDataSource("xpoDataSource", xpoDataSource.SaveToXml())
		End Sub
	End Class
End Namespace

File: CS/MvcDashboardDataSources/Configuration/XpoDataSourceConfigurator.cs
using DevExpress.DashboardCommon;
using DevExpress.DashboardWeb;

namespace MvcDashboardDataSources.Configuration {
    public class XpoDataSourceConfigurator {
        public static void ConfigureDataSource(DataSourceInMemoryStorage storage) {
            // Registers an XPO data source.
            DashboardXpoDataSource xpoDataSource = new DashboardXpoDataSource("XPO Data Source");
            xpoDataSource.ConnectionStringName = "NWindConnectionStringSQLite";
            xpoDataSource.SetEntityType(typeof(Category));
            storage.RegisterDataSource("xpoDataSource", xpoDataSource.SaveToXml());
        }
    }
}